### PR TITLE
[DEV-1017] Configure nextjs to manage CMS contents

### PR DESCRIPTION
**Pull request by MarcoPonchia**
#### List of Changes

Nextjs manage content recovery from CMS

#### Motivation and Context

Nextjs can retrive data from CMS

#### How Has This Been Tested?

Manually user can:
- see static content if `CMS_BASE_PATH` is not present in .env.local
- see CMS content if `CMS_BASE_PATH="http://localhost:1337/api"` is defined in .env.local, to proprely test this part [this PR](https://github.com/pagopa/developer-portal/pull/599) must be merged or user have to manually merge in the same branch

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .changeset/wild-pugs-wash.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"nextjs-website": major
```
Comment (marcobottaro) [inline .changeset/wild-pugs-wash.md:2]:
```suggestion
"nextjs-website": minor
```

File: apps/nextjs-website/src/lib/types/homepageResponseData.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export type HomepageApi = {
+  readonly comingsoonDocumentation: {
+    readonly title: string;
+    readonly links: readonly {
+      readonly text: string;
+      readonly href: string;
+    }[];
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/types/homepageResponseData.ts:7]: Add a TODO to manage `target` attribute

**Review by marcobottaro — CHANGES_REQUESTED** [review]

**Review by marcobottaro — APPROVED** [review]

**Comment by MarcoPonchia** [comment]
This PR has been converted to draft because a direct comparison is needed to define the refactor together using "fp-ts"

File: apps/nextjs-website/src/lib/api.ts [hunk 3]
```diff
@@ -90,6 +112,41 @@ export async function getProducts(): Promise<readonly Product[]> {
...
+    productsShowcaseTitle: homepage.productsShowcaseTitle,
+    comingsoonDocumentation: {
+      title: homepage.comingsoonDocumentation.title,
+      links: homepage.comingsoonDocumentation.links,
+    },
+  };
+  if (!usingCms) return staticContent;
+
+  const response = await cmsRequest<HomepageApi>(
+    'home-page',
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/api.ts:137]:
```suggestion
    'homepage',
```

**Comment by MarcoPonchia** [comment]
This task has been refactor and polished here: https://github.com/pagopa/developer-portal/pull/615

**Pull request closed without merging**